.github/workflows/build-containers.yml

```diff
@@ -0,0 +1,51 @@
+name: Build Containers
+
+on: [workflow_dispatch]
+
+jobs:
+  list-containers:
+    name: List containers
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      - uses: actions/checkout@v2
+
+      - id: set-matrix
+        working-directory: ./devops/docker
+        run: echo "::set-output name=matrix::$(ls **/Dockerfile | jq -R -s -c 'split("\n")[:-1]')"
+
+  docker:
```

Comment:
Is there the ability to target specific distros? It looks like all containers are built when we might be onboarding a new distro, without us needing to rebuild all of the environments. Consider adding options so we can selectively build any container. Similar to what we currently have in Azure Pipelines
![image](https://user-images.githubusercontent.com/224804/163596008-477be7d5-8e47-4a65-a59c-a5bd3bcf2e91.png)

Reply:
Updated to only build the containers containing changes merged to `main`. Any `Dockerfile` in `./devops/docker/` will be built and push if it is added, copied, modified, or renamed by PR that was merged to main.